javascript/linked-list/__tests__/linked-list.test.js

```diff
@@ -30,4 +30,31 @@ describe('linked list test suite', () => {
       '{ 1 } -> { a } -> { b } -> { c } -> NULL'
     );
   });
+  test('append can insert a node at the end of the list', () => {
+    //insert a node then check that the last node of the list matches a value
+    // insert it
+    // compare it to our expected toString value
+    const testLL = new LinkedList(dummyLL);
+    testLL.append('d');
+    expect(testLL.toString()).toEqual(
+      '{ a } -> { b } -> { c } -> { d } -> NULL'
+    );
+  });
+  //option 2--------------
+  //traverse list
+  // if current.next = null return current expect current.value to === the value we inserted
+  test('Can succesfully insert a node before a node located in the middle of a linked list', () => {
+    const testLLBefore = new LinkedList(dummyLL); //instantiate new list every time an option
```

Comment:
You are not instantiating a new list every time because you are passing `dummyLL` by reference instead of making a new copy. The one way to deal with this could be 
```js
const testLLBefore = new LinkedList({...dummyLL});